### PR TITLE
make travis run tests

### PR DESCRIPTION
**Pull request by diracdeltas**
and add some test coverage for siteUtil when sync is enabled

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ before_install:
   - export DISPLAY=:99.0
   - sh -e /etc/init.d/xvfb start
 script:
-  - npm run lint && npm test
+  - npm run lint && npm run download-sync-client && npm test
```
Comment (posix4e) [inline .travis.yml]: Why are these connected via && instead of multiple - lines?

File: tools/downloadSyncClient.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
```
Comment (posix4e) [inline tools/downloadSyncClient.sh]: I actually think the new normal is to name things tools/downloadSyncClient not tools/downloadSyncClient.sh unless you really are a .sh script (not bash)

File: tools/downloadSyncClient.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+# stop on errors (nonzero exit codes), uninitialized vars
+set -eu
+
+SYNC_PATH=app/extensions/brave/content/scripts/sync.js
+SYNC_URL=https://raw.githubusercontent.com/brave/sync-bundles/master/bundle.js
+
+curl -o $SYNC_PATH $SYNC_URL
```
Comment (ayumi) [inline tools/downloadSyncClient.sh]: I added bundle.js and testbundle.js from brave/sync-bundles as the first release of brave/sync.

Download latest prod (non pre-release) version:
```sh
curl -O $(curl -s https://api.github.com/repos/brave/sync/releases/latest | grep \/bundle.js | head -n 1 | cut -d '"' -f 4)
```

File: .travis.yml (hunk 1 above)
Comment (diracdeltas) [inline .travis.yml]: dunno

**Comment by diracdeltas** [comment]
made requested changes

**Comment by diracdeltas** [comment]
travis still seems broken for reasons unrelated to this change, looking at that now

**Comment by diracdeltas** [comment]
fixing the other travis error now

**Comment by diracdeltas** [comment]
travis is now actually running tests

**Comment by diracdeltas** [comment]
travis then stalled on the tests and gave up, but merging for now since it's some progress

**Pull request merged**